definitions/procedures/backup/config_files.rb

```diff
@@ -16,17 +17,45 @@ class ConfigFiles < ForemanMaintain::Procedure
     end
 
     def run
-      tarball = File.join(@backup_dir, 'config_files.tar.gz')
-      increments = File.join(@backup_dir, '.config.snar')
       with_spinner('Collecting config files to backup') do
-        configs = config_files.join(' ')
-        statuses = @ignore_changed_files ? [0, 1] : [0]
-        execute!("tar --selinux --create --gzip --file=#{tarball} " \
-          "--listed-incremental=#{increments} --ignore-failed-read " \
-          "#{configs}", :valid_exit_statuses => statuses)
+        create_tarball
       end
     end
 
+    def create_tarball
+      (1..MAX_RETRIES).each do |ret|
+        exit_status = run_tar_cmd
+        break unless statuses.include? exit_status
+
+        warn "\nRemoving config files archive #{tarball} as its incomplete"
```

Comment:
Are we going to show warnings on every run if one fails?

As I haven't tested it but I would like to see difference between two outputs i.e using `spinner.update`  and using `warn`.

Reply:
The current output with warn and warn!,

~~~
# bin/foreman-maintain advanced procedure run backup-config-files --backup-dir /tmp/
Running ForemanMaintain::Scenario
================================================================================
Backup config files: 
/ Collecting config files to backup                                             
Removing config files archive /tmp/config_files.tar.gz as its incomplete
Recollecting config files backup, retry 1 !
\ Collecting config files to backup                                             
Removing config files archive /tmp/config_files.tar.gz as its incomplete
Recollecting config files backup, retry 2 !
/ Collecting config files to backup                                             
Removing config files archive /tmp/config_files.tar.gz as its incomplete
Recollecting config files backup, retry 3 !
~~~


* If I use the spinner.update for "Removing config files archive /tmp/config_files.tar.gz as its incomplete"

~~~
# bin/foreman-maintain advanced procedure run backup-config-files --backup-dir /tmp/
Running ForemanMaintain::Scenario
================================================================================
Backup config files: 
\                                                                               
Removing config files archive /tmp/config_files.tar.gz as its incompleteRecollecting config files backup, retry 1 !
\                                                                               
|                                                                               
Removing config files archive /tmp/config_files.tar.gz as its incompleteRecollecting config files backup, retry 2 !
|                                                                               
/                                                                               
-                                                                               
Removing config files archive /tmp/config_files.tar.gz as its incompleteRecollecting config files backup, retry 3 !
~~~

* As per this I thought of using it for "Recollecting config files backup, retry 3 !"

~~~
# bin/foreman-maintain advanced procedure run backup-config-files --backup-dir /tmp/
Running ForemanMaintain::Scenario
================================================================================
Backup config files: 
-                                                                               
-                                                                               
-                                                                               
\                                                                               
|                                                                               
|                                                                               
|                                                                               
/                                                                               
-                                                                               
-                                                                               
Recollecting config files backup, retry 3 !                           [FAIL]
~~~

* One more thing here is if I use spinner.update instead of warn! when MAX_RETRIES== ret ; the run show status as OK instead of FAILED, I tried to forcefully fail it by setting @status= :fail, @status =1 but it did not help to make it fail. If I am missing anything here then request to elaborate on same ?

~~~
warn! 'Config files backup failed' if MAX_RETRIES == ret
~~~

* I preferred to show warning at each run as its good to understand that we are deleting the failed backup and retrying again, so user is not confused anywhere.